File: backend/dto/lost_report.py
# backend/app/schemas/lost_report.py
from pydantic import BaseModel, field_validator
from datetime import datetime
from typing import Optional


class LostReportCreate(BaseModel):
    lat: float
    lng: float
    message: Optional[str] = None


class LostReportStatusUpdate(BaseModel):
    """Обновление статуса сообщения (админ)"""
    status: str  # pending, accepted, completed, cancelled

    @field_validator('status')
    @classmethod
    def validate_status(cls, v):
        allowed = ['pending', 'accepted', 'completed', 'cancelled']
        if v not in allowed:
            raise ValueError(f'status must be one of: {allowed}')
        return v


class LostReportResponse(BaseModel):
    report_id: int
    user_id: int
    user_nickname: str
    lat: float
    lng: float
    message: Optional[str] = None
    status: str  # pending, accepted, completed, cancelled
    created_at: datetime

    class Config:
        from_attributes = True

File: мобилка/app/src/main/java/com/sologo/app/domain/model/LostReport.kt
package com.sologo.app.domain.model

import java.util.Date

data class LostReport(
    val id: Int,
    val userId: Int,
    val userNickname: String,
    val lat: Double,
    val lng: Double,
    val message: String?,
    val status: LostStatus,
    val createdAt: Date
)

enum class LostStatus {
    PENDING, ACCEPTED, COMPLETED, CANCELLED;

    companion object {
        fun fromString(status: String): LostStatus {
            return when (status.lowercase()) {
                "accepted" -> ACCEPTED
                "completed" -> COMPLETED
                "cancelled" -> CANCELLED
                else -> PENDING
            }
        }
    }
}